client/cli/src/commands/utils.rs

```diff
@@ -274,15 +274,17 @@ where
 }
 
 /// checks if message is Some, otherwise reads message from stdin and optionally decodes hex
-pub fn read_message(msg: Option<&String>, should_decode: bool) -> Result<Vec<u8>, Error> {
+pub fn read_message(msg: Option<&String>, decode_hex: bool) -> Result<Vec<u8>, Error> {
 	let mut message = vec![];
 	match msg {
 		Some(m) => {
-			message = array_bytes::hex2bytes(m.as_str())?;
+			if decode_hex {
+				message = array_bytes::hex2bytes(m.as_str())?;
+			}
```

Comment:
If `decode_hex == false`, you will not have any message?

If you want to pass `--message binary_data` we would need to change `message` to `Vec<u8>` otherwise the binary data would be required to be valid utf8 data.

Reply:
Yea… having binary data as CLI arg is rather uncommon. Clap interprets `Vec<u8>` not as raw data but as list of `u8`.  
So i will have it interpret `--message` as UTF8-non hex per default, as opposed to UTF8-hex before.